classes/order/OrderHistory.php

```diff
@@ -616,4 +602,40 @@ public function addWs()
             return $this->add();
         }
     }
+
+
+    /**
+     * Creates a payment for an order and optionally associates it with an invoice.
+     *
+     * @param Order $order The order object associated with the payment.
+     * @param float $restPaid The remaining amount to be paid.
+     * @param int $invoiceId The ID of the invoice to associate the payment with.
+     *
+     * @return void
+     */
+    protected function createOrderPayment(Order $order, float $restPaid, int $invoiceId = 0): void {
+        if ($restPaid <= 0) {
+            return;
+        }
+        $payment = new OrderPayment();
+        $payment->order_reference = Tools::substr($order->reference, 0, 9);
+        $payment->id_currency = $order->id_currency;
+        $payment->amount = $restPaid;
+        $payment->payment_method = isset($payment_method) && $payment_method instanceof Module ? $payment_method->displayName : null;
```

Comment:
It seems that is not correct code since `$payment_method` variable is not available. Overall, there are a few errors reported by PHP Stan that should help you fix the code, so that we can proceed to the review and validation.

Thanks @Jeremie-Kiwik !